lib/handler/mruby/channel.c

```diff
@@ -73,7 +56,7 @@ static mrb_value channel_initialize_method(mrb_state *mrb, mrb_value self)
     memset(ctx, 0, sizeof(*ctx));
     assert(shared_ctx->current_context != NULL);
     ctx->ctx = shared_ctx->current_context;
-    ctx->receiver = mrb_nil_value();
+    ctx->receivers = mrb_ary_new(mrb);
```

Comment:
`receivers` should be `mrb_gc_register` ed?

Reply:
Thanks, I see. Do you think it should be `mrb_gc_register ` or `mrb_gc_protect` after shited also?

Reply:
Sorry I don't understand your question. mrb_gc_registerとmrb_gc_protectのどちらを使うべきかという意味ですか？であれば、ここではregisterのほうですね。 (protectだとgc arena indexがrestoreされたらgcされてしまうので）
それともshiftされたobjectも保護すべきかどうか、という質問ですかね？ (shited => shifted?)

Reply:
Sorry for confusing.
shiftされたobjectも保護すべきかどうかという質問です。

Reply:
Is it enough because mrb_gc_arena_save is done here?
https://github.com/naritta/h2o/blob/916cab15f8f88cfb608b769379f6af4139fb8612/lib/handler/mruby/channel.c#L74-L77

Reply:
Ah, I think you have to call `mrb_gc_protect(mrb, shifted_receiver)` immediately after shifting from receivers. Otherwise it can be GCed in mrb_funcall function before `call` method of that Proc is called.

Reply:
Thanks, I see. I fixed to call `mrb_gc_protect ` here.
https://github.com/h2o/h2o/pull/1336/commits/8926f7d74423c852f6ab10e58fda90e0fda6aee6